docs/apworld_dev_faq.md

```diff
@@ -43,3 +43,26 @@ A faster alternative to the `for` loop would be to use a [list comprehension](ht
 ```py
 item_pool += [self.create_filler() for _ in range(total_locations - len(item_pool))]
 ```
+
+---
+
+### I learned about indirect conditions in the world API document, but I want to know more. What are they and why are they necessary?
+
+The world API document mentions how to use `multiworld.register_indirect_condition` to register indirect conditions and **when** you should use them, but not *how* they work and *why* they are necessary. This is because the explanation is quite complicated.
+
+Region sweep (the algorithm that determines which regions are reachable) is a Breadth-First Search of the region graph from the origin region, checking entrances one by one and adding newly reached nodes (regions) and their entrances to the queue until there is nothing more to check.
+
+For performance reasons, AP only checks every entrance once. However, if any entrance's access_rule depends on region access, then it is possible for this to happen:
+1. An entrance that depends on a region is checked and determined to be nontraversable because the region hasn't been reached yet during the graph search.
+2. After that, the region is reached by the graph search.
+
+The entrance *would* now be determined to be traversable if it were rechecked, but it is not.
+To account for this case, AP would have to recheck all entrances every time a new region is reached until no new regions are reached.
+
+However, there is a way to **manually** define that a *specific* entrance needs to be rechecked during region sweep if a *specific* region is reached during it. This is what an indirect condition is.
```

Comment:
```suggestion
An indirect condition is how you can **manually** define that a *specific* entrance needs to be rechecked during region sweep if a *specific* region is reached during it.
```
and idk about the \*\*, up to you

Reply:
all italics/bold formatting was written by vi and i have am ambivalent to, if you think it makes it read worse suggest them removed and I will